plugins/node/instrumentation-tedious/src/instrumentation.ts

```diff
@@ -0,0 +1,226 @@
+        `TediousInstrumentation: patched Connection.prototype.${operation}`
+      );
+
+      function patchedMethod(this: ApproxConnection, request: ApproxRequest) {
+        let procCount = 0;
+        let statementCount = 0;
+        const incrementStatementCount = () => statementCount++;
+        const incrementProcCount = () => procCount++;
+        const databaseName = this[CURRENT_DATABASE];
+        const sql = (request => {
+          // Required for <11.0.9
+          if (
+            request.sqlTextOrProcedure === 'sp_prepare' &&
+            request?.parametersByName?.stmt?.value
+          ) {
+            return request.parametersByName.stmt.value;
+          }
+          return request.sqlTextOrProcedure;
+        })(request);
+
+        const span = thisPlugin.tracer.startSpan(
+          getSpanName(operation, databaseName, sql, request.table),
+          {
+            kind: api.SpanKind.CLIENT,
+            attributes: {
+              [SemanticAttributes.DB_SYSTEM]: DbSystemValues.MSSQL,
+              [SemanticAttributes.DB_NAME]: databaseName,
+              [SemanticAttributes.NET_PEER_PORT]: this.config?.options?.port,
+              [SemanticAttributes.NET_PEER_NAME]: this.config?.server,
+              // >=4 uses `authentication` object, older versions just userName and password pair
+              [SemanticAttributes.DB_USER]:
+                this.config.userName ??
+                this.config?.authentication?.options?.userName,
+              [SemanticAttributes.DB_STATEMENT]: sql,
+              [SemanticAttributes.DB_SQL_TABLE]: request.table,
+            },
+          }
+        );
+
+        const endSpan = once((err?: any) => {
+          request.removeListener('done', incrementStatementCount);
+          request.removeListener('doneInProc', incrementStatementCount);
+          request.removeListener('doneProc', incrementProcCount);
+          request.removeListener('error', endSpan);
+          this.removeListener('end', endSpan);
+
+          span.setAttribute('tedious.procedure_count', procCount);
+          span.setAttribute('tedious.statement_count', statementCount);
+          if (err) {
+            span.setStatus({
+              code: api.SpanStatusCode.ERROR,
+              message: err.message,
+            });
+          }
+          span.end();
+        });
+
+        request.on('done', incrementStatementCount);
+        request.on('doneInProc', incrementStatementCount);
+        request.on('doneProc', incrementProcCount);
+        request.once('error', endSpan);
+        this.on('end', endSpan);
+
+        if (typeof request.callback === 'function') {
+          thisPlugin._wrap(
+            request,
+            'callback',
+            thisPlugin._patchCallbackQuery(endSpan)
+          );
+        }
+
+        return api.context.with(
+          api.trace.setSpan(api.context.active(), span),
+          originalMethod,
+          this,
+          ...arguments
+        );
```

Comment:
Potentially yes:

- DNS checks,
- FS calls(not instrumented today),
- TCP instrumentation.

I'd say it's a bug, we don't do it everywhere. Howcome we expect that instrumentations in contrib are the leaf instrumentations?

Reply:
It depend of each instrumentation, but i would say that the rule would be that if it can perform remote call, it should be set in the context so that the remote span is a child of this action.